### ISSUE TITLE
Packaging docker for Mageia

### ISSUE THREAD

**Issue by bcornec** [docker/docker#9453]
I'm packaging docker for Mageia. At least I've been successful with the 1.2.x branch, but less with the 1.3 one.

Mageia is an RPM based distro. I've started my package looking at the Fedora one, and then derived the work for Mageia. I also added the various golang packages that were needed in order to support docker and docker-registry.

Since version 1.3.x I have been unable to build correctly docker. It seems to be linked to my ignorance of the go language for sure, and also to the usage of external go modules instead of the one you provide in your tar.gz.

For reference my spec file is attached.
Latest working version spec file (for 1.2.x) is at http://svnweb.mageia.org/packages/cauldron/docker-io/current/SPECS/docker-io.spec?revision=745282&view=markup

The error msgs I get with that version are also attached in the log file.

I do not seem to be able to use your embedded version of libcontainer providing the syncpipe stuff (which is now removed from the latest libcontainer. I tried to add it in the mageia package (providing the 3 files removed without more luck).

BTW I do not seem to be able to build nsinit in libcontainer either, which is leading ot other issues.

I'm fighting since the issue of 1.3.0 to add docker to Mageia, but I'm not making enough progresses here, so I'm calling for hint !

Let me know what you'd like me to try.

You can also use the official mageia docker container to try, or mine bcornec/mageia to see whether you can reproduce.

My goal would be to minimize the usage of embedded go packages (maintenance aspects and security) that I'd prefer to patch with appropriate content to support docker, rather than using embedded versions. Now as a start I'd be just happy to make it work, which is not the case, even using all embedded versions (I tried that as well).

Each time I get different error msgs, from Go, despite trying to put the right GOPATH which should provide me with your versions.

Thanks in advance for any help you could provide and for your work around Docker.

[-- Attachement #1: docker-io.spec --]
```
# modifying the dockerinit binary breaks the SHA1 sum check by docker
%global __os_install_post %{_rpmconfigdir}/brp-compress

#debuginfo not supported with Go
%global debug_package %{nil}
%global import_path github.com/docker/docker
%global gopath  %{_libdir}/golang
%define gosrc %{gopath}/src/pkg/%{import_path}

# for 1.3.2
%global commit      39fa2faad2f3d6fa5133de4eb740677202f53ef4
%global shortcommit %(c=%{commit}; echo ${c:0:7})

Name:           docker-io
#Version:        1.3.1.git%{shortcommit}
Version:        1.3.2
Release:        %mkrel 1
Summary:        Automates deployment of containerized applications
License:        ASL 2.0
Group:                  System/Configuration

URL:            http://www.docker.com
# only x86_64 for now: https://github.com/docker/docker/issues/136
ExclusiveArch:  x86_64
Source0:        https://%{import_path}/archive/v%{version}.tar.gz
#Source0:        https://%{import_path}/archive/%{commit}.tar.gz
BuildRequires:  gcc
BuildRequires:  glibc-static-devel

# ensure build uses golang 1.2-7 and above
# http://code.google.com/p/go/source/detail?r=a15f344a9efa35ef168c8feaa92a15a1cdc93db5
BuildRequires:  golang >= 1.3.3
BuildRequires:  libsqlite3-devel
BuildRequires:  golang(github.com/gorilla/mux)
BuildRequires:  golang(github.com/gorilla/context)
BuildRequires:  golang(github.com/kr/pty)
BuildRequires:  golang(github.com/godbus/dbus)
BuildRequires:  golang(github.com/coreos/go-systemd) >= 0-0.4
BuildRequires:  golang(code.google.com/p/go.net/websocket)
BuildRequires:  golang(code.google.com/p/gosqlite/sqlite3)
BuildRequires:  golang(github.com/syndtr/gocapability/capability)
BuildRequires:  golang(github.com/docker/libcontainer)
BuildRequires:  golang(github.com/tchap/go-patricia/patricia)
BuildRequires:  golang(github.com/docker/libtrust)
BuildRequires:  golang(github.com/docker/libtrust/trustgraph)
BuildRequires:  go-md2man
BuildRequires:  device-mapper-devel
BuildRequires:  btrfs-devel
BuildRequires:  systemd-devel
# Build upstream docs with pandoc
#BuildRequires:  pandoc
Requires:       systemd-units

# need xz to work with ubuntu images
# https://bugzilla.redhat.com/show_bug.cgi?id=1045220
Requires:       xz

# https://bugzilla.redhat.com/show_bug.cgi?id=1035436
# this won't be needed for rhel7+
Requires:       bridge-utils
Requires:       lxc

# https://bugzilla.redhat.com/show_bug.cgi?id=1034919
# No longer needed in Fedora because of libcontainer
Requires:       libcgroup
Provides:       lxc-docker = %{version}
# Can't have it with docker as use conflicting binary names
Conflicts:      docker

%description
Docker is an open-source engine that automates the deployment of any
application as a lightweight, portable, self-sufficient container that will
run virtually anywhere.

Docker containers can encapsulate any payload, and will run consistently on
and between virtually any server. The same container that a developer builds
and tests on a laptop will run at scale, in production*, on VMs, bare-metal
servers, OpenStack clusters, public instances, or combinations of the above.

%package devel
BuildRequires:  golang >= 1.3.3
Requires:       golang >= 1.3.3
Requires:       docker-io-pkg-devel
Summary:        A golang registry for global request variables (source libraries)
Provides:       golang(%{import_path}) = %{version}-%{release}
Provides:       golang(%{import_path}/api) = %{version}-%{release}
Provides:       golang(%{import_path}/api/client) = %{version}-%{release}
Provides:       golang(%{import_path}/api/server) = %{version}-%{release}
Provides:       golang(%{import_path}/archive) = %{version}-%{release}
Provides:       golang(%{import_path}/builtins) = %{version}-%{release}
Provides:       golang(%{import_path}/contrib) = %{version}-%{release}
Provides:       golang(%{import_path}/contrib/docker-device-tool) = %{version}-%{release}
Provides:       golang(%{import_path}/contrib/host-integration) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/execdriver) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/execdriver/execdrivers) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/execdriver/lxc) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/execdriver/native) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/execdriver/native/configuration) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/execdriver/native/template) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/graphdriver) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/graphdriver/aufs) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/graphdriver/btrfs) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/graphdriver/devmapper) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/graphdriver/graphtest) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/graphdriver/vfs) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/networkdriver) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/networkdriver/bridge) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/networkdriver/ipallocator) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/networkdriver/portallocator) = %{version}-%{release}
Provides:       golang(%{import_path}/daemon/networkdriver/portmapper) = %{version}-%{release}
Provides:       golang(%{import_path}/dockerversion) = %{version}-%{release}
Provides:       golang(%{import_path}/engine) = %{version}-%{release}
Provides:       golang(%{import_path}/graph) = %{version}-%{release}
Provides:       golang(%{import_path}/image) = %{version}-%{release}
Provides:       golang(%{import_path}/integration) = %{version}-%{release}
Provides:       golang(%{import_path}/integration-cli) = %{version}-%{release}
Provides:       golang(%{import_path}/links) = %{version}-%{release}
Provides:       golang(%{import_path}/nat) = %{version}-%{release}
Provides:       golang(%{import_path}/opts) = %{version}-%{release}
Provides:       golang(%{import_path}/registry) = %{version}-%{release}
Provides:       golang(%{import_path}/runconfig) = %{version}-%{release}
Provides:       golang(%{import_path}/utils) = %{version}-%{release}
Provides:       golang(%{import_path}/utils/broadcastwriter) = %{version}-%{release}

%description devel
This is the source libraries for docker.

%package pkg-devel
BuildRequires:  golang >= 1.3.3
Requires:       golang >= 1.3.3
Summary:        A golang registry for global request variables (source libraries)
Provides:       golang(%{import_path}/pkg/graphdb) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/iptables) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/listenbuffer) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/mflag) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/mflag/example) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/mount) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/namesgenerator) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/networkfs/etchosts) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/networkfs/resolvconf) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/proxy) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/signal) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/symlink) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/sysinfo) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/system) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/systemd) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/tailfile) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/term) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/testutils) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/truncindex) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/units) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/user) = %{version}-%{release}
Provides:       golang(%{import_path}/pkg/version) = %{version}-%{release}

%description pkg-devel
These source librariees are provided by docker, but are independent of docker specific logic.
The import paths of %{import_path}/pkg/...
%prep
#%setup -q -n docker-%{commit}
%setup -q -n docker-%{version}
#rm -rf vendor/src/code.google.com vendor/src/github.com/{coreos,docker/libtrust,godbus,gorilla,kr,syndtr,tchap}
#for f in `find . -name '*.go'`; do
        #perl -pi -e 's|github.com/docker/docker/vendor/src/code.google.com/p/go/src/pkg/archive/tar|archive/tar|' $f
#done

%build
mkdir -p _build

pushd _build
  mkdir -p src/github.com/docker
  ln -s $(dirs +1 -l) src/github.com/docker/docker
popd

export DOCKER_GITCOMMIT="%{shortcommit}/%{version}"
export GOPATH=$(pwd)/_build:$(pwd)/vendor:%{gopath}

hack/make.sh dynbinary
docs/man/md/md2man-all.sh
cp contrib/syntax/vim/LICENSE LICENSE-vim-syntax
cp contrib/syntax/vim/README.md README-vim-syntax.md

%install
# install binary
install -d %{buildroot}%{_bindir}
install -p -m 755 bundles/%{version}/dynbinary/docker-%{version} %{buildroot}%{_bindir}/docker

# install dockerinit
install -d %{buildroot}%{_libexecdir}/docker
install -p -m 755 bundles/%{version}/dynbinary/dockerinit-%{version} %{buildroot}%{_libexecdir}/docker/dockerinit

# Place to store images
install -d %{buildroot}%{_libexecdir}/cache/docker

# install manpages
install -d %{buildroot}%{_mandir}/man1
install -p -m 644 docs/man/man1/docker*.1 %{buildroot}%{_mandir}/man1
install -d %{buildroot}%{_mandir}/man5
install -p -m 644 docs/man/man5/Dockerfile.5 %{buildroot}%{_mandir}/man5

# install bash completion
install -d %{buildroot}%{_sysconfdir}/bash_completion.d
install -p -m 644 contrib/completion/bash/docker %{buildroot}%{_sysconfdir}/bash_completion.d/docker.bash
# install zsh completion
install -d %{buildroot}%{_datadir}/zsh/site-functions
install -p -m 644 contrib/completion/zsh/_docker %{buildroot}%{_datadir}/zsh/site-functions
# install vim syntax highlighting
install -d %{buildroot}%{_datadir}/vim/vimfiles/{doc,ftdetect,syntax}
install -p -m 644 contrib/syntax/vim/doc/dockerfile.txt %{buildroot}%{_datadir}/vim/vimfiles/doc
install -p -m 644 contrib/syntax/vim/ftdetect/dockerfile.vim %{buildroot}%{_datadir}/vim/vimfiles/ftdetect
install -p -m 644 contrib/syntax/vim/syntax/dockerfile.vim %{buildroot}%{_datadir}/vim/vimfiles/syntax
# install udev rules
install -d %{buildroot}%{_udevrulesdir}
install -p -m 755 contrib/udev/80-docker.rules %{buildroot}%{_udevrulesdir}
# install storage dir
install -d -m 700 %{buildroot}%{_sharedstatedir}/docker
# install systemd/init scripts
install -d %{buildroot}%{_unitdir}
install -p -m 644 contrib/init/systemd/docker.service %{buildroot}%{_unitdir}

# It's convenient to have docker listening on a tcp port so add it
# the fd: param doesn't seem to work as well as the unix one
# Also use rather /var/cache to store images
perl -pi -e 's|-H fd://|-H unix://var/run/docker.sock -H tcp://127.0.0.1:2375 -g /var/cache/docker|'  %{buildroot}%{_unitdir}/docker.service
# Fix rights on docker socket so it can start - systemd should do it, but it's systemd you know !
perl -pi -e "s|^SocketUser|#Doesn't seem to work\nSocketUser|m" %{buildroot}%{_unitdir}/docker.socket
perl -pi -e "s|^SocketGroup=docker|SocketGroup=docker\n# So do it another way\nExecStartPost=/usr/bin/chgrp docker /var/run/docker.sock|m"
%{buildroot}%{_unitdir}/docker.socket

# sources
install -d -p %{buildroot}/%{gosrc}

for dir in api archive builtins daemon dockerversion engine graph \
           image links nat opts pkg registry runconfig utils
do
        cp -rpav $dir %{buildroot}/%{gosrc}
done

%pre
getent group docker > /dev/null || %{_sbindir}/groupadd -r docker
exit 0

%post
%systemd_post docker

%preun
%systemd_preun docker

%postun
%systemd_postun_with_restart docker

%files
%doc AUTHORS CHANGELOG.md CONTRIBUTING.md LICENSE MAINTAINERS NOTICE README.md
%doc LICENSE-vim-syntax README-vim-syntax.md
%{_mandir}/man1/docker*.1.gz
%{_mandir}/man5/Dockerfile.5.gz
%{_bindir}/docker
%dir %{_libexecdir}/docker
%dir %{_libexecdir}/cache/docker
%{_libexecdir}/docker/dockerinit
%{_unitdir}/docker.service
%{_unitdir}/docker.socket
%dir %{_sysconfdir}/bash_completion.d
%{_sysconfdir}/bash_completion.d/docker.bash
%{_datadir}/zsh/site-functions/_docker
%dir %{_sharedstatedir}/docker
%dir %{_udevrulesdir}
%{_udevrulesdir}/80-docker.rules
%dir %{_datadir}/vim/vimfiles/doc
%{_datadir}/vim/vimfiles/doc/dockerfile.txt
%dir %{_datadir}/vim/vimfiles/ftdetect
%{_datadir}/vim/vimfiles/ftdetect/dockerfile.vim
%dir %{_datadir}/vim/vimfiles/syntax
%{_datadir}/vim/vimfiles/syntax/dockerfile.vim

%files devel
%dir %{gosrc}
%dir %{gosrc}/api
%{gosrc}/api/MAINTAINERS
%{gosrc}/api/README.md
%{gosrc}/api/*.go
%dir %{gosrc}/api/client
%{gosrc}/api/client/*.go
%dir %{gosrc}/api/server
%{gosrc}/api/server/MAINTAINERS
%{gosrc}/api/server/*.go
%dir %{gosrc}/archive
%{gosrc}/archive/MAINTAINERS
%{gosrc}/archive/README.md
%{gosrc}/archive/*.go
%dir %{gosrc}/archive/testdata
%{gosrc}/archive/testdata/broken.tar
%dir %{gosrc}/builtins
%{gosrc}/builtins/*.go
%dir %{gosrc}/daemon
%{gosrc}/daemon/*.go
%{gosrc}/daemon/MAINTAINERS
%{gosrc}/daemon/README.md
%dir %{gosrc}/daemon/execdriver
%{gosrc}/daemon/execdriver/*.go
%{gosrc}/daemon/execdriver/MAINTAINERS
%dir %{gosrc}/daemon/execdriver/execdrivers
%{gosrc}/daemon/execdriver/execdrivers/*.go
%dir %{gosrc}/daemon/execdriver/lxc
%{gosrc}/daemon/execdriver/lxc/MAINTAINERS
%{gosrc}/daemon/execdriver/lxc/*.go
%dir %{gosrc}/daemon/execdriver/native
%{gosrc}/daemon/execdriver/native/*.go
%dir %{gosrc}/daemon/execdriver/native/configuration
%{gosrc}/daemon/execdriver/native/configuration/*.go
%dir %{gosrc}/daemon/execdriver/native/template
%{gosrc}/daemon/execdriver/native/template/*.go
%dir %{gosrc}/daemon/graphdriver
%{gosrc}/daemon/graphdriver/*.go
%dir %{gosrc}/daemon/graphdriver/aufs
%{gosrc}/daemon/graphdriver/aufs/*.go
%dir %{gosrc}/daemon/graphdriver/btrfs
%{gosrc}/daemon/graphdriver/btrfs/*.go
%{gosrc}/daemon/graphdriver/btrfs/MAINTAINERS
%dir %{gosrc}/daemon/graphdriver/devmapper
%{gosrc}/daemon/graphdriver/devmapper/*.go
%{gosrc}/daemon/graphdriver/devmapper/MAINTAINERS
%{gosrc}/daemon/graphdriver/devmapper/README.md
%dir %{gosrc}/daemon/graphdriver/graphtest
%{gosrc}/daemon/graphdriver/graphtest/*.go
%dir %{gosrc}/daemon/graphdriver/vfs
%{gosrc}/daemon/graphdriver/vfs/*.go
%dir %{gosrc}/daemon/networkdriver
%dir %{gosrc}/daemon/networkdriver/bridge
%{gosrc}/daemon/networkdriver/bridge/*.go
%dir %{gosrc}/daemon/networkdriver/ipallocator
%{gosrc}/daemon/networkdriver/ipallocator/*.go
%{gosrc}/daemon/networkdriver/*.go
%dir %{gosrc}/daemon/networkdriver/portallocator
%{gosrc}/daemon/networkdriver/portallocator/*.go
%dir %{gosrc}/daemon/networkdriver/portmapper
%{gosrc}/daemon/networkdriver/portmapper/*.go
%dir %{gosrc}/dockerversion
%{gosrc}/dockerversion/*.go
%dir %{gosrc}/engine
%{gosrc}/engine/MAINTAINERS
%{gosrc}/engine/*.go
%dir %{gosrc}/graph
%{gosrc}/graph/MAINTAINERS
%{gosrc}/graph/*.go
%dir %{gosrc}/image
%{gosrc}/image/*.go
%dir %{gosrc}/links
%{gosrc}/links/*.go
%dir %{gosrc}/nat
%{gosrc}/nat/*.go
%dir %{gosrc}/opts
%{gosrc}/opts/*.go
%dir %{gosrc}/registry
%{gosrc}/registry/MAINTAINERS
%{gosrc}/registry/*.go
%dir %{gosrc}/runconfig
%{gosrc}/runconfig/*.go
%dir %{gosrc}/utils
%{gosrc}/utils/*.go

%files pkg-devel
%dir %{gosrc}
%dir %{gosrc}/pkg
%{gosrc}/pkg/README.md
%dir %{gosrc}/pkg/broadcastwriter
%{gosrc}/pkg/broadcastwriter/*.go
%dir %{gosrc}/pkg/graphdb
%{gosrc}/pkg/graphdb/MAINTAINERS
%{gosrc}/pkg/graphdb/*.go
%dir %{gosrc}/pkg/httputils
%{gosrc}/pkg/httputils/MAINTAINERS
%{gosrc}/pkg/httputils/*.go
%dir %{gosrc}/pkg/iptables
%{gosrc}/pkg/iptables/MAINTAINERS
%{gosrc}/pkg/iptables/*.go
%dir %{gosrc}/pkg/jsonlog
%{gosrc}/pkg/jsonlog/*.go
%dir %{gosrc}/pkg/listenbuffer
%{gosrc}/pkg/listenbuffer/*.go
%dir %{gosrc}/pkg/log
%{gosrc}/pkg/log/*.go
%dir %{gosrc}/pkg/mflag
%{gosrc}/pkg/mflag/LICENSE
%{gosrc}/pkg/mflag/MAINTAINERS
%{gosrc}/pkg/mflag/README.md
%dir %{gosrc}/pkg/mflag/example
%{gosrc}/pkg/mflag/example/example.go
%{gosrc}/pkg/mflag/*.go
%dir %{gosrc}/pkg/mount
%{gosrc}/pkg/mount/MAINTAINERS
%{gosrc}/pkg/mount/*.go
%dir %{gosrc}/pkg/namesgenerator
%{gosrc}/pkg/namesgenerator/*.go
%dir %{gosrc}/pkg/networkfs
%{gosrc}/pkg/networkfs/MAINTAINERS
%dir %{gosrc}/pkg/networkfs/etchosts
%{gosrc}/pkg/networkfs/etchosts/*.go
%dir %{gosrc}/pkg/networkfs/resolvconf
%{gosrc}/pkg/networkfs/resolvconf/*.go
%dir %{gosrc}/pkg/parsers
%{gosrc}/pkg/parsers/MAINTAINERS
%{gosrc}/pkg/parsers/*.go
%dir %{gosrc}/pkg/parsers/filters
%{gosrc}/pkg/parsers/filters/*.go
%dir %{gosrc}/pkg/parsers/kernel
%{gosrc}/pkg/parsers/kernel/*.go
%dir %{gosrc}/pkg/parsers/operatingsystem
%{gosrc}/pkg/parsers/operatingsystem/*.go
%dir %{gosrc}/pkg/proxy
%{gosrc}/pkg/proxy/MAINTAINERS
%{gosrc}/pkg/proxy/*.go
%dir %{gosrc}/pkg/signal
%{gosrc}/pkg/signal/*.go
%dir %{gosrc}/pkg/symlink
%{gosrc}/pkg/symlink/MAINTAINERS
%{gosrc}/pkg/symlink/*.go
%dir %{gosrc}/pkg/symlink/testdata
%dir %{gosrc}/pkg/symlink/testdata/fs
%dir %{gosrc}/pkg/symlink/testdata/fs/a
%{gosrc}/pkg/symlink/testdata/fs/a/d
%{gosrc}/pkg/symlink/testdata/fs/a/e
%{gosrc}/pkg/symlink/testdata/fs/a/f
%dir %{gosrc}/pkg/symlink/testdata/fs/b
%{gosrc}/pkg/symlink/testdata/fs/b/h
%{gosrc}/pkg/symlink/testdata/fs/g
%{gosrc}/pkg/symlink/testdata/fs/i
%dir %{gosrc}/pkg/sysinfo
%{gosrc}/pkg/sysinfo/MAINTAINERS
%{gosrc}/pkg/sysinfo/*.go
%dir %{gosrc}/pkg/system
%{gosrc}/pkg/system/MAINTAINERS
%{gosrc}/pkg/system/*.go
%dir %{gosrc}/pkg/systemd
%{gosrc}/pkg/systemd/MAINTAINERS
%{gosrc}/pkg/systemd/*.go
%dir %{gosrc}/pkg/tailfile
%{gosrc}/pkg/tailfile/*.go
%dir %{gosrc}/pkg/tarsum
%{gosrc}/pkg/tarsum/*.go
%dir %{gosrc}/pkg/tarsum/testdata
%dir %{gosrc}/pkg/tarsum/testdata/46af0962ab5afeb5ce6740d4d91652e69206fc991fd5328c1a94d364ad00e457
%{gosrc}/pkg/tarsum/testdata/46af0962ab5afeb5ce6740d4d91652e69206fc991fd5328c1a94d364ad00e457/json
%{gosrc}/pkg/tarsum/testdata/46af0962ab5afeb5ce6740d4d91652e69206fc991fd5328c1a94d364ad00e457/layer.tar
%dir %{gosrc}/pkg/tarsum/testdata/511136ea3c5a64f264b78b5433614aec563103b4d4702f3ba7d4d2698e22c158
%{gosrc}/pkg/tarsum/testdata/511136ea3c5a64f264b78b5433614aec563103b4d4702f3ba7d4d2698e22c158/json
%{gosrc}/pkg/tarsum/testdata/511136ea3c5a64f264b78b5433614aec563103b4d4702f3ba7d4d2698e22c158/layer.tar
%dir %{gosrc}/pkg/truncindex
%{gosrc}/pkg/truncindex/MAINTAINERS
%{gosrc}/pkg/truncindex/*.go
%dir %{gosrc}/pkg/term
%{gosrc}/pkg/term/MAINTAINERS
%{gosrc}/pkg/term/*.go
%dir %{gosrc}/pkg/testutils
%{gosrc}/pkg/testutils/MAINTAINERS
%{gosrc}/pkg/testutils/README.md
%{gosrc}/pkg/testutils/utils.go
%dir %{gosrc}/pkg/units
%{gosrc}/pkg/units/MAINTAINERS
%{gosrc}/pkg/units/*.go
%dir %{gosrc}/pkg/version
%{gosrc}/pkg/version/*.go
```

[-- Attachement #2: log.docker-io --]
```
Exécution_de(%prep) : /bin/sh -e /users/bruno/prj/mageia/docker-io/BUILDROOT/rpm-tmp.vYMiCC
+ umask 022
+ cd /users/bruno/prj/mageia/docker-io/BUILD
+ '[' 1 -eq 1 ']'
+ '[' 1 -eq 1 ']'
+ '[' 1 -eq 1 ']'
+ cd /users/bruno/prj/mageia/docker-io/BUILD
+ rm -rf docker-1.3.2
+ /usr/bin/gzip -dc /users/bruno/prj/mageia/docker-io/SOURCES/v1.3.2.tar.gz
+ /usr/bin/tar -xf -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd docker-1.3.2
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ exit 0
Exécution_de(%build) : /bin/sh -e /users/bruno/prj/mageia/docker-io/BUILDROOT/rpm-tmp.i2IIRZ
+ umask 022
+ cd /users/bruno/prj/mageia/docker-io/BUILD
+ cd docker-1.3.2
+ '[' 1 -eq 1 ']'
+ '[' 1 -eq 1 ']'
+ mkdir -p _build
+ pushd _build
~/prj/mageia/docker-io/BUILD/docker-1.3.2/_build ~/prj/mageia/docker-io/BUILD/docker-1.3.2
+ mkdir -p src/github.com/docker
++ dirs +1 -l
+ ln -s /users/bruno/prj/mageia/docker-io/BUILD/docker-1.3.2 src/github.com/docker/docker
+ popd
~/prj/mageia/docker-io/BUILD/docker-1.3.2
+ export DOCKER_GITCOMMIT=39fa2fa/1.3.2
+ DOCKER_GITCOMMIT=39fa2fa/1.3.2
++ pwd
++ pwd
+ export
GOPATH=/users/bruno/prj/mageia/docker-io/BUILD/docker-1.3.2/_build:/users/bruno/prj/mageia/docker-io/BUILD/docker-1.3.2/vendor:/usr/lib64/go
lang
+
GOPATH=/users/bruno/prj/mageia/docker-io/BUILD/docker-1.3.2/_build:/users/bruno/prj/mageia/docker-io/BUILD/docker-1.3.2/vendor:/usr/lib64/go
lang
+ hack/make.sh dynbinary
# WARNING! I don't seem to be running in the Docker container.
# The result of this command might be an incorrect build, and will not be
#   officially supported.
#
# Try this instead: make all
#

---> Making bundle: dynbinary (in bundles/1.3.2/dynbinary)
# github.com/docker/docker/daemon/execdriver/native
_build/src/github.com/docker/docker/daemon/execdriver/native/init.go:56: cannot use syncPipe (type *syncpipe.SyncPipe) as type *os.File in
argument to namespaces.Init
erreur : Mauvais statut de sortie pour /users/bruno/prj/mageia/docker-io/BUILDROOT/rpm-tmp.i2IIRZ (%build)


Erreur de construction de RPM :
    Mauvais statut de sortie pour /users/bruno/prj/mageia/docker-io/BUILDROOT/rpm-tmp.i2IIRZ (%build)
```

**Comment by bcornec** [comment]
Hello,

I'm back on this as Mageia is preparing the 5 release, and I'd really like to have docker in.

The Dockerfile update for 1.4.1 still works:

Step 4 : ENV DOCKER_GITCOMMIT 5bc2ff8a36e9a768e8b479de4fe3ea9c9daf4121
 ---> Running in 01ac309a12a4
 ---> d54db114074e
Removing intermediate container 01ac309a12a4
Step 5 : RUN AUTO_GOPATH=1 ./hack/make.sh dynbinary
 ---> Running in 54250dbaf800
# WARNING! I don't seem to be running in the Docker container.
# The result of this command might be an incorrect build, and will not be                                                                   
#   officially supported.                                                                                                                   
#                                                                                                                                           
# Try this instead: make all                                                                                                                
#                                                                                                                                           

---> Making bundle: dynbinary (in bundles/1.4.1/dynbinary)
Created binary: /tmp/docker-1.4.1/bundles/1.4.1/dynbinary/dockerinit-1.4.1
Created binary: /tmp/docker-1.4.1/bundles/1.4.1/dynbinary/docker-1.4.1

 ---> 93aa6e0d4847
Removing intermediate container 54250dbaf800
Successfully built 93aa6e0d4847

However, trying the same build sequence for the RPM still fails, even if I aligned now you build instructions to what I do in the spec file:

Exécution_de(%build) : /bin/sh -e /users/bruno/prj/mageia/docker/BUILDROOT/rpm-tmp.mkNRXg
+ umask 022
+ cd /users/bruno/prj/mageia/docker/BUILD
+ cd docker-1.4.1
+ '[' 1 -eq 1 ']'
+ '[' 1 -eq 1 ']'
+ export DOCKER_GITCOMMIT=5bc2ff8
+ DOCKER_GITCOMMIT=5bc2ff8
+ AUTO_GOPATH=1
+ ./hack/make.sh dynbinary
# WARNING! I don't seem to be running in the Docker container.
# The result of this command might be an incorrect build, and will not be
#   officially supported.
#
# Try this instead: make all
#

---> Making bundle: dynbinary (in bundles/1.4.1/dynbinary)
# github.com/docker/docker/pkg/archive
.gopath/src/github.com/docker/docker/pkg/archive/changes.go:138: undefined: system.Stat
erreur : Mauvais statut de sortie pour /users/bruno/prj/mageia/docker/BUILDROOT/rpm-tmp.mkNRXg (%build)

So I still have one dependency locally which is different from the one I get in the container ! (However both are using golang 1.3.3-2)

I just have more golang packages in my environment that what you have:
golang-sqlite-0.1.git74691fb6f837-2.mga4
golang-testify-0.git8ce79b9f0b77-2.mga4
golang-codegangsta-devel-1.2.0-1.mga4
golang-syndtr-gocapability-0.1.git3454319-2.mga4
golang-libtrust-0.1.git9f61d45-2.mga4
golang-gopatricia-1.0.1-1.mga4
golang-libcontainer-1.2.0.git7ce34f5-7.mga4
golang-gorilla-context-0.1.gita08edd3-2.mga4
golang-testify-devel-0.git8ce79b9f0b77-2.mga4
golang-go-systemd-2-3.mga4
golang-kr-pty-0.1.git3b1f648-2.mga4
golang-net-0.1.git84a4013f96e0-2.mga4
golang-logrus-0.6.0-1.mga4
golang-logrus-devel-0.6.0-1.mga4
golang-1.3.3-2.mga4
golang-vim-1.3.2-1.mga4
golang-blackfriday-1.2-1.mga4
golang-godbus-1-2.mga4
golang-gorilla-mux-0.1.git9ede152-2.mga4
golang-libcontainer-devel-1.2.0.git7ce34f5-7.mga4

Could one of these take precedence over what I have in the vendor dir ?

**Comment by bcornec** [comment]
If I remove all my local golang packages in order to only get what you provide I have the following:

+ export DOCKER_GITCOMMIT=5bc2ff8
+ DOCKER_GITCOMMIT=5bc2ff8
+ AUTO_GOPATH=1
+ ./hack/make.sh dynbinary
# WARNING! I don't seem to be running in the Docker container.
# The result of this command might be an incorrect build, and will not be
#   officially supported.
#
# Try this instead: make all
#

---> Making bundle: dynbinary (in bundles/1.4.1/dynbinary)
.gopath/src/github.com/docker/docker/daemon/execdriver/native/template/default_template.go:4:2: no buildable Go source files in /usr/lib64/golang/src/pkg/github.com/docker/libcontainer
.gopath/src/github.com/docker/docker/daemon/execdriver/lxc/driver.go:18:2: no buildable Go source files in /usr/lib64/golang/src/pkg/github.com/kr/pty

So it seems indeed related to an environment aspect, but I'm still unsure what I need to keep and what I need to use from your delivery, even if your docker build upper would sugeest I need nothing special

**Comment by tianon** [comment]
Ah, if you're packaging the dependencies instead of using the `vendor/` directory as-is, you probably don't want to use `AUTO_GOPATH`, since then it won't reference your system-installed versions (especially if you've pruned the `vendor/` directory from the tarball).

**Comment by bcornec** [comment]
Well my original goal was to build all dependencies separately as individual packages indeed. But since this was failing, I'm now back to try to be nearer from what you do, so using AUTOGO. But even in that case that doesn't work building as a package while it works inyour docker env:

Example with GOPATH:
+ mkdir -p _build
+ pushd _build
~/prj/mageia/docker/BUILD/docker-1.4.1/_build ~/prj/mageia/docker/BUILD/docker-1.4.1
+ mkdir -p src/github.com/docker
++ dirs +1 -l
+ ln -s /users/bruno/prj/mageia/docker/BUILD/docker-1.4.1 src/github.com/docker/docker
+ popd
~/prj/mageia/docker/BUILD/docker-1.4.1
+ export DOCKER_GITCOMMIT=5bc2ff8
+ DOCKER_GITCOMMIT=5bc2ff8
++ pwd
++ pwd
+ export GOPATH=/users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/_build:/users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/vendor:/usr/lib64/golang
+ GOPATH=/users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/_build:/users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/vendor:/usr/lib64/golang
+ hack/make.sh dynbinary
# WARNING! I don't seem to be running in the Docker container.
# The result of this command might be an incorrect build, and will not be
#   officially supported.
#
# Try this instead: make all
#

---> Making bundle: dynbinary (in bundles/1.4.1/dynbinary)
_build/src/github.com/docker/docker/daemon/execdriver/native/template/default_template.go:4:2: no buildable Go source files in /usr/lib64/golang/src/pkg/github.com/docker/libcontainer
_build/src/github.com/docker/docker/daemon/execdriver/lxc/driver.go:18:2: no buildable Go source files in /usr/lib64/golang/src/pkg/github.com/kr/pty


Example with AUTOGO:
+ export DOCKER_GITCOMMIT=5bc2ff8
+ DOCKER_GITCOMMIT=5bc2ff8
+ AUTO_GOPATH=1
+ ./hack/make.sh dynbinary
# WARNING! I don't seem to be running in the Docker container.
# The result of this command might be an incorrect build, and will not be
#   officially supported.
#
# Try this instead: make all
#

---> Making bundle: dynbinary (in bundles/1.4.1/dynbinary)
.gopath/src/github.com/docker/docker/daemon/execdriver/native/template/default_template.go:4:2: no buildable Go source files in /usr/lib64/g
olang/src/pkg/github.com/docker/libcontainer
.gopath/src/github.com/docker/docker/daemon/execdriver/lxc/driver.go:18:2: no buildable Go source files in /usr/lib64/golang/src/pkg/github.com/kr/pty

So I fully nderstand it's lined to my env, but Go is the first language I'm involved with to package which is creating such a nightmare wrt native build :-( And reading pages such as http://golang.org/cmd/go/#hdr-GOPATH_environment_variable

I just want to understand how can I specify to Go where are the files it needs to use to do its include mechanism. like the -I of C e.g. GOPATH is poiting to the place where the sources are, with the right subtree, but doesn't find them:

> ls BUILD/docker-1.4.1/vendor/src/github.com/docker/libcontainer/
api_temp.go     console          error.go     LICENSE               namespaces  PRINCIPLES.md   security  types.go
apparmor        container.go     factory.go   MAINTAINERS           netlink     process.go      selinux   update-vendor.sh
cgroups         CONTRIBUTING.md  integration  MAINTAINERS_GUIDE.md  network     README.md       SPEC.md   user
config.go       devices          ipc          Makefile              NOTICE      ROADMAP.md      state.go  utils
config_test.go  Dockerfile       label        mount                 nsinit      sample_configs  system    xattr

I'm really puzzled :-(

**Comment by tianon** [comment]
It's really bizarre that even with `AUTO_GOPATH=1`, Go is looking in `/usr/lib64/golang/src/pkg/github.com/kr/pty` to find `github.com/kr/pty`. :cry:

It's going to be very, very verbose, but could you please try `bash -x ./hack/make.sh dynbinary` (with `export AUTO_GOPATH=1`) so we can hopefully see where it's tripping up?

BTW, when you're posting code blocks or other output, you should enclose it in triple-backticks (see https://guides.github.com/features/mastering-markdown/#examples; click on "Code" there to see what I mean) so that it's easier to read and doesn't get other markdown formatting inadvertently.  I've taken the liberty of adjusting the comments above already so I could read them easier. :angel:

**Comment by bcornec** [comment]
Thanks for your help Tianon, and sorry for the format issue.

Here is the log with bash -x:

```
+ export DOCKER_GITCOMMIT=5bc2ff8
+ DOCKER_GITCOMMIT=5bc2ff8
+ AUTO_GOPATH=1
+ bash -x ./hack/make.sh dynbinary
+ set -e
+ set -o pipefail
+ export DOCKER_PKG=github.com/docker/docker
+ DOCKER_PKG=github.com/docker/docker
++ pwd
+ '[' /users/bruno/prj/mageia/docker/BUILD/docker-1.4.1 '!=' /go/src/github.com/docker/docker ']'
+ echo '# WARNING! I don'\''t seem to be running in the Docker container.'
# WARNING! I don't seem to be running in the Docker container.
+ echo '# The result of this command might be an incorrect build, and will not be'
# The result of this command might be an incorrect build, and will not be
+ echo '#   officially supported.'
#   officially supported.
+ echo '#'
#
+ echo '# Try this instead: make all'
# Try this instead: make all
+ echo '#'
#
+ echo

+ DEFAULT_BUNDLES=(validate-dco validate-gofmt binary test-unit test-integration test-integration-cli dynbinary dyntest-unit dyntest-integration cover cross tgz ubuntu)
++ cat ./VERSION
+ VERSION=1.4.1
+ command -v git
+ git rev-parse
+ '[' 5bc2ff8 ']'
+ GITCOMMIT=5bc2ff8
+ '[' 1 ']'
+ rm -rf .gopath
++ dirname github.com/docker/docker
+ mkdir -p .gopath/src/github.com/docker
+ ln -sf ../../../.. .gopath/src/github.com/docker/docker
++ pwd
++ pwd
+ export GOPATH=/users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/.gopath:/users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/vendor
+ GOPATH=/users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/.gopath:/users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/vendor
+ '[' '!' /users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/.gopath:/users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/vendor ']'
+ '[' -z '' ']'
+ DOCKER_BUILDTAGS+=' daemon'
+ LDFLAGS='
        -X github.com/docker/docker/dockerversion.GITCOMMIT "5bc2ff8"
        -X github.com/docker/docker/dockerversion.VERSION "1.4.1"
'
+ '[' -z '' ']'
+ LDFLAGS='-w 
        -X github.com/docker/docker/dockerversion.GITCOMMIT "5bc2ff8"
        -X github.com/docker/docker/dockerversion.VERSION "1.4.1"
'
+ LDFLAGS_STATIC='-linkmode external'
++ go env GOOS
+ '[' linux == windows ']'
+ EXTLDFLAGS_STATIC=-static
+ ORIG_BUILDFLAGS=(-a -tags "netgo static_build $DOCKER_BUILDTAGS")
+ BUILDFLAGS=($BUILDFLAGS "${ORIG_BUILDFLAGS[@]}")
+ : 30m
+ TESTFLAGS+=' -test.timeout=30m'
+ EXTLDFLAGS_STATIC_DOCKER='-static -lpthread -Wl,--unresolved-symbols=ignore-in-object-files'
+ LDFLAGS_STATIC_DOCKER='
        -linkmode external
        -X github.com/docker/docker/dockerversion.IAMSTATIC true
        -extldflags "-static -lpthread -Wl,--unresolved-symbols=ignore-in-object-files"
'
++ uname -s
+ '[' Linux = FreeBSD ']'
+ '[' '!' -e /usr/include/sqlite3.h ']'
+ HAVE_GO_TEST_COVER=
+ go help testflag
+ grep -- -cover
+ go tool -n cover
+ main dynbinary
+ mkdir -p bundles
+ '[' -e bundles/1.4.1 ']'
+++ dirname ./hack/make.sh
++ cd ./hack
++ pwd
+ SCRIPTDIR=/users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/hack
+ '[' 1 -lt 1 ']'
+ bundles=($@)
+ for bundle in '${bundles[@]}'
+ bundle /users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/hack/make/dynbinary
+ bundlescript=/users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/hack/make/dynbinary
++ basename /users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/hack/make/dynbinary
+ bundle=dynbinary
+ echo '---> Making bundle: dynbinary (in bundles/1.4.1/dynbinary)'
---> Making bundle: dynbinary (in bundles/1.4.1/dynbinary)
+ mkdir -p bundles/1.4.1/dynbinary
++ pwd
+ source /users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/hack/make/dynbinary /users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/bundles/1.4.1/dynbinary
++ set -e
++ DEST=/users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/bundles/1.4.1/dynbinary
++ '[' -z '' ']'
++ go build -o /users/bruno/prj/mageia/docker/BUILD/docker-1.4.1/bundles/1.4.1/dynbinary/dockerinit-1.4.1 -a -tags 'netgo static_build  daemon' -ldflags '
                        -w 
        -X github.com/docker/docker/dockerversion.GITCOMMIT "5bc2ff8"
        -X github.com/docker/docker/dockerversion.VERSION "1.4.1"

                        -linkmode external
                        -extldflags "-static"
                ' ./dockerinit
.gopath/src/github.com/docker/docker/daemon/execdriver/native/template/default_template.go:4:2: no buildable Go source files in /usr/lib64/golang/src/pkg/github.com/docker/libcontainer
.gopath/src/github.com/docker/docker/daemon/execdriver/lxc/driver.go:18:2: no buildable Go source files in /usr/lib64/golang/src/pkg/github.com/kr/pty

**Comment by tianon** [comment]
It looks like it might be picking up some old state in `/usr/lib64/golang/src/`.  Can you do a `find /usr/lib64/golang/src/pkg/github.com` and paste the output here?  Even if there are just empty directories in there, that could cause what we're seeing (my hunch is that `go env GOROOT` will be `/usr/lib64/golang`, and Go will prefer packages that already exist in the `GOROOT` over anything anywhere else, including and especially from the `GOPATH`, so installing anything besides Go itself there is a really bad idea).

**Comment by bcornec** [comment]
Right there is remaining content:
```
find /usr/lib64/golang/src/pkg/github.com
/usr/lib64/golang/src/pkg/github.com
/usr/lib64/golang/src/pkg/github.com/tchap
/usr/lib64/golang/src/pkg/github.com/tchap/go-patricia
/usr/lib64/golang/src/pkg/github.com/syndtr
/usr/lib64/golang/src/pkg/github.com/syndtr/gocapability
/usr/lib64/golang/src/pkg/github.com/coreos
/usr/lib64/golang/src/pkg/github.com/Sirupsen
/usr/lib64/golang/src/pkg/github.com/russross
/usr/lib64/golang/src/pkg/github.com/docker
/usr/lib64/golang/src/pkg/github.com/docker/docker
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/log
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/log/log.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/log/log_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/units
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/units/duration.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/units/size_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/units/duration_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/units/size.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/units/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/system
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/system/utimes_darwin.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/system/utimes_unsupported.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/system/stat_unsupported.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/system/utimes_linux.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/system/utimes_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/system/xattrs_linux.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/system/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/system/utimes_freebsd.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/system/xattrs_unsupported.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/system/errors.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/system/stat_linux.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/fs_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/testdata
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/testdata/fs
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/testdata/fs/b
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/testdata/fs/b/h
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/testdata/fs/g
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/testdata/fs/i
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/testdata/fs/a
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/testdata/fs/a/e
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/testdata/fs/a/f
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/testdata/fs/a/d
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/symlink/fs.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tailfile
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tailfile/tailfile_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tailfile/tailfile.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/iptables
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/iptables/iptables.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/iptables/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/namesgenerator
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/namesgenerator/names-generator.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/namesgenerator/names-generator_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/networkfs
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/networkfs/resolvconf
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/networkfs/resolvconf/resolvconf_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/networkfs/resolvconf/resolvconf.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/networkfs/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/networkfs/etchosts
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/networkfs/etchosts/etchosts_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/networkfs/etchosts/etchosts.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/jsonlog
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/jsonlog/jsonlog.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tarsum
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tarsum/tarsum.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tarsum/tarsum_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tarsum/testdata
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tarsum/testdata/46af0962ab5afeb5ce6740d4d91652e69206fc991fd5328c1a94d364ad00e457
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tarsum/testdata/46af0962ab5afeb5ce6740d4d91652e69206fc991fd5328c1a94d364ad00e457/json
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tarsum/testdata/46af0962ab5afeb5ce6740d4d91652e69206fc991fd5328c1a94d364ad00e457/layer.tar
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tarsum/testdata/511136ea3c5a64f264b78b5433614aec563103b4d4702f3ba7d4d2698e22c158
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tarsum/testdata/511136ea3c5a64f264b78b5433614aec563103b4d4702f3ba7d4d2698e22c158/json
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/tarsum/testdata/511136ea3c5a64f264b78b5433614aec563103b4d4702f3ba7d4d2698e22c158/layer.tar
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/sysinfo
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/sysinfo/sysinfo.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/sysinfo/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/README.md
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/listenbuffer
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/listenbuffer/buffer.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/truncindex
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/truncindex/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/truncindex/truncindex.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/truncindex/truncindex_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/parsers.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/parsers_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/filters
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/filters/parse_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/filters/parse.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/operatingsystem
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/operatingsystem/operatingsystem.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/operatingsystem/operatingsystem_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/kernel
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/kernel/uname_unsupported.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/kernel/uname_linux.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/kernel/kernel.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/parsers/kernel/kernel_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/broadcastwriter
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/broadcastwriter/broadcastwriter.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/broadcastwriter/broadcastwriter_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mflag
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mflag/LICENSE
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mflag/flag.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mflag/example
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mflag/example/example.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mflag/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mflag/README.md
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mflag/flag_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/signal
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/signal/trap.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/signal/signal_darwin.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/signal/signal.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/signal/signal_unsupported.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/signal/signal_linux.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/signal/signal_freebsd.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/testutils
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/testutils/utils.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/testutils/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/testutils/README.md
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/httputils
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/httputils/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/httputils/resumablerequestreader.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/proxy
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/proxy/tcp_proxy.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/proxy/udp_proxy.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/proxy/network_proxy_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/proxy/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/proxy/proxy.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/proxy/stub_proxy.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/term
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/term/term.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/term/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/term/termios_linux.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/term/termios_darwin.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/term/termios_freebsd.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/systemd
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/systemd/booted.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/systemd/sd_notify.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/systemd/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/systemd/listendfd.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/graphdb
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/graphdb/conn_unsupported.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/graphdb/utils.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/graphdb/conn_sqlite3.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/graphdb/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/graphdb/graphdb.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/graphdb/sort_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/graphdb/graphdb_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/graphdb/sort.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/mountinfo.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/mountinfo_linux_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/MAINTAINERS
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/flags_linux.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/mountinfo_linux.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/flags_freebsd.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/mount.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/mount_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/flags.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/mounter_linux.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/mountinfo_freebsd.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/mounter_freebsd.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/mountinfo_unsupported.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/flags_unsupported.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/mount/mounter_unsupported.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/version
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/version/version_test.go
/usr/lib64/golang/src/pkg/github.com/docker/docker/pkg/version/version.go
/usr/lib64/golang/src/pkg/github.com/docker/libcontainer
/usr/lib64/golang/src/pkg/github.com/docker/libcontainer/security
/usr/lib64/golang/src/pkg/github.com/codegangsta
/usr/lib64/golang/src/pkg/github.com/godbus
/usr/lib64/golang/src/pkg/github.com/kr
/usr/lib64/golang/src/pkg/github.com/kr/pty
/usr/lib64/golang/src/pkg/github.com/stretchr
/usr/lib64/golang/src/pkg/github.com/cpuguy83
/usr/lib64/golang/src/pkg/github.com/cpuguy83/go-md2man
/usr/lib64/golang/src/pkg/github.com/cpuguy83/go-md2man/mangen
/usr/lib64/golang/src/pkg/github.com/cpuguy83/go-md2man/mangen/mangen.go
/usr/lib64/golang/src/pkg/github.com/cpuguy83/go-md2man/md2man.go
/usr/lib64/golang/src/pkg/github.com/gorilla
/usr/lib64/golang/src/pkg/github.com/gorilla/context
/usr/lib64/golang/src/pkg/github.com/gorilla/mux
```

Some were belonging to the package docker-io-pkg-devel-1.2.0-6.mga4, others to the package  go-md2man-1-1.mga4.x86_64 which are now uninstalled to avoid issues.

The rest has been removed manually:
find /usr/lib64/golang/src/pkg/github.com
find: ‘/usr/lib64/golang/src/pkg/github.com’: No such file or directory

Now trying to build docker goes way further ! The remaining problems are mine at that point.

Thanks a lot for tracking that down. Is there anything I could do to prevent go to look under that part of the tree where there are just empty dirs and no content ? I'm surprised by that behaviour, comparing to other languages. Could it be considered as a bug somehow ?

When I tried to package the dependencies separately I installed them in that place and some dirs where not removed when I uninstalled the packages thus the issue you found. 

My next step will be to build docker as a package with the full content you provide and avoid for now using separate packages (That's what Fedora does however).

**Comment by tianon** [comment]
I would say don't ever install files there unless they're in the `golang` package itself.  Period.  Full-stop.  Go upstream certainly won't support you if you're injecting other non-stdlib packages in the stdlib directory. :wink:

**Comment by bcornec** [comment]
So where are the additional go packages supposed to be installed then ? For perl, python we don't have the type of problem. I need blackfriday and net and for now they are here:
```
> rpm -ql golang-blackfriday-devel
/usr/lib64/golang/src/pkg/github.com/russross/blackfriday
/usr/lib64/golang/src/pkg/github.com/russross/blackfriday/block.go
[...]
/usr/lib64/golang/src/pkg/github.com/russross/blackfriday/upskirtref_test.go
/usr/share/doc/golang-blackfriday-devel
/usr/share/doc/golang-blackfriday-devel/README.md
> rpm -ql golang-net-devel
/usr/lib64/golang/src/pkg/code.google.com/p/go.net/codereview.cfg
/usr/lib64/golang/src/pkg/code.google.com/p/go.net/dict
/usr/lib64/golang/src/pkg/code.google.com/p/go.net/dict/dict.go
[...]
/usr/lib64/golang/src/pkg/code.google.com/p/go.net/websocket/websocket_test.go
/usr/share/doc/golang-net-devel
/usr/share/doc/golang-net-devel/AUTHORS
/usr/share/doc/golang-net-devel/CONTRIBUTORS
/usr/share/doc/golang-net-devel/LICENSE
/usr/share/doc/golang-net-devel/PATENTS
/usr/share/doc/golang-net-devel/README
```
Any pointer to a doc on this you could recommend ?
Thanks again for your help Tianon, much appreciated.

BTW I've now a working docker package 1.4.1 for Mageia !

**Comment by tianon** [comment]
For packaging, you should install extra packages to some other path (and make sure it's a valid `GOPATH`), and then have your Go packages set that `GOPATH` appropriately.  Debian handles this reasonably well (although not perfect); see https://wiki.debian.org/MichaelStapelberg/GoPackaging#Why_should_I_use_.2BIBw-go_get.2BIB0_instead_of_apt-get_to_install_Go_libraries.3F for more info (specifically, they use a `GOPATH` of `/usr/share/gocode`, and then `dh_golang` is responsible for setting `GOPATH` when building a new Go package).

**Comment by unclejack** [comment]
@tianon Should this stay open?

**Comment by bcornec** [comment]
No we can close it. Mageia 5 will have docker and docker-registry. I used docker's deliveries instead of trying to build a proper stack. Hopefully that will be possible one day when bugs have been solved in the other pakages.
